src/status_im/ui/screens/ens/events.cljs

```diff
@@ -0,0 +1,123 @@
+
+(defn- assoc-username [db username]
+  (assoc-in db [:ens :username] username))
+
+(defn- valid-custom-domain? [username]
+  (and (ens/is-valid-eth-name? username)
+       (stateofus/lower-case? username)))
+
+(defn- valid-username? [custom-domain? username]
+  (if custom-domain?
+    (valid-custom-domain? username)
+    (stateofus/valid-username? username)))
+
+(handlers/register-handler-fx
+ :ens/set-state
+ (fn [{:keys [db]} [_ state]]
+   {:db (assoc-state db state)}))
+
+(defn- state [valid? username]
+  (cond
+    (string/blank? username) :initial
+    valid? :typing
+    :else
+    :invalid))
+
+(handlers/register-handler-fx
+ :ens/set-username
+ (fn [{:keys [db]} [_ custom-domain? username]]
+   (let [valid? (valid-username? custom-domain? username)]
+     (merge
+      {:db (-> db
+               (assoc-username username)
+               (assoc-state (state valid? username)))}
+      (when valid?
+        (let [{:keys [account/account]}        db
+              {:keys [address public-key]}     account
+              registry (get ens/ens-registries (chain db))
+              name     (if custom-domain? username (stateofus/subdomain username))]
+          {::ens/resolve [registry name #(on-resolve registry custom-domain? name address public-key %)]}))))))
+
+(handlers/register-handler-fx
+ :ens/navigate-back
+ (fn [{:keys [db] :as cofx} _]
+   (fx/merge cofx
+             {:db (-> db
+                      (assoc-state :initial)
+                      (assoc-username ""))}
+             (navigation/navigate-back))))
+
+(handlers/register-handler-fx
+ :ens/switch-domain-type
+ (fn [{:keys [db]} _]
+   {:db (-> (update-in db [:ens :custom-domain?] not)
+            (assoc-state :initial)
+            (assoc-username ""))}))
+
+(handlers/register-handler-fx
+ :ens/on-registration-success
+ (fn [{:keys [db]} _]
+   {:db       (assoc-state db :registered)
+    :dispatch [:navigate-back]}))
+
+(handlers/register-handler-fx
+ :ens/on-registration-failure
+ (fn [{:keys [db]} _]
+   {:db (assoc-state db :registration-failed)}))
+
+(defn- prepare-extension-transaction [{:keys [contract username address public-key]}]
+  (let [method        "register(bytes32,address,bytes32,bytes32)"
+        {:keys [x y]} (ethereum/coordinates public-key)
+        data          (abi-spec/encode method [(ethereum/sha3 username) address x y])]
+    {:to-name         "Stateofus registrar"
+     :symbol          :ETH
+     :method          method
+     :amount          (money/bignumber 0)
+     :to              contract
+     :gas             (money/bignumber 200000)
+     :data            data
+     :on-result       [:ens/on-registration-success]
+     :on-error        [:ens/on-registration-failure]}))
+
+(handlers/register-handler-fx
+ :ens/register
+ (fn [{db :db} [_ data]]
+   (let [transaction (prepare-extension-transaction data)]
+     ;; Last argument is nil: no need to estimate gas
+     (wallet/open-modal-wallet-for-transaction db transaction nil))))
```

Comment:
use `wallet/eth-transaction-call` here

Reply:
Can't find this function.

Reply:
It's in my PR which should get merged today